packages/wrangler/src/publish.ts

```diff
@@ -220,21 +220,25 @@ export default async function publish(props: Props): Promise<void> {
 	// TODO: warn if git/hg has uncommitted changes
 	const { config, accountId, name } = props;
 	if (accountId && name) {
-		const serviceMetaData = await fetchResult(
-			`/accounts/${accountId}/workers/services/${name}`
-		);
-		const { default_environment } = serviceMetaData as {
-			default_environment: {
-				script: { last_deployed_from: "dash" | "wrangler" | "api" };
+		try {
+			const serviceMetaData = await fetchResult(
+				`/accounts/${accountId}/workers/services/${name}`
+			);
+			const { default_environment } = serviceMetaData as {
+				default_environment: {
+					script: { last_deployed_from: "dash" | "wrangler" | "api" };
+				};
 			};
-		};
 
-		if (
-			(await fromDashMessagePrompt(
-				default_environment.script.last_deployed_from
-			)) === false
-		)
-			return;
+			if (
+				(await fromDashMessagePrompt(
+					default_environment.script.last_deployed_from
+				)) === false
+			)
+				return;
+		} catch (e) {
+			// ignore errors
```

Comment:
this should only be silent if it's the service not found error

Reply:
Done.